src/pages/ScoreReport.vue

```diff
@@ -883,150 +911,197 @@ const viewOptions = ref([
+        tableRow[`${taskName} - Reliability`] = 'Reliable';
       }
 
-      if (score.reliable !== undefined && !score.reliable && score.engagementFlags !== undefined) {
-        const engagementFlags = Object.keys(score.engagementFlags);
-        if (engagementFlags.length > 0) {
-          if (includedValidityFlags[taskId]) {
-            const filteredFlags = Object.keys(score.engagementFlags).filter((flag) =>
-              includedValidityFlags[taskId].includes(flag),
-            );
-            if (filteredFlags.length === 0) {
-              tableRow[`${tasksDictionary.value[taskId]?.publicName ?? taskId} - Reliability`] = 'Unreliable';
-            } else {
-              const engagementFlagString = 'Unreliable: ' + filteredFlags.map((key) => _lowerCase(key)).join(', ');
-              tableRow[`${tasksDictionary.value[taskId]?.publicName ?? taskId} - Reliability`] = engagementFlagString;
+      // Add progress immediately after reliability if includeProgress is true
+      if (includeProgress) {
+        const progressRow = computedProgressData.value.find(
+          (progress) => progress.userPid === _get(user, 'assessmentPid'),
+        );
+
+        if (progressRow) {
+          scoreReportColumns.value.forEach((column) => {
+            const { field, header: taskName } = column; // Use taskName from the column header
+
+            // Ensure field is defined and is a string before calling startsWith
+            if (field && typeof field === 'string' && field.startsWith('scores')) {
+              const scoreKey = field.split('.').slice(-2, -1)[0]; // Extract taskId (e.g., "swr", "sre", etc.)
+
+              // Check if taskId exists in progressRow.progress
+              if (progressRow.progress[scoreKey]) {
+                tableRow[`${taskName} - Progress`] = progressRow.progress[scoreKey].value;
+              } else {
+                tableRow[`${taskName} - Progress`] = 'not assigned';
+              }
             }
-          } else {
-            const engagementFlagString = 'Unreliable: ' + engagementFlags.map((key) => _lowerCase(key)).join(', ');
-            tableRow[`${tasksDictionary.value[taskId]?.publicName ?? taskId} - Reliability`] = engagementFlagString;
-          }
+          });
         } else {
-          tableRow[`${tasksDictionary.value[taskId]?.publicName ?? taskId} - Reliability`] = 'Assessment Incomplete';
+          // If no progressRow is found, mark all scores as "not assigned"
+          scoreReportColumns.value.forEach((column) => {
+            const { field, header: taskName } = column; // Use taskName from the column header
+
+            // Ensure field is defined and is a string before calling startsWith
+            if (field && typeof field === 'string' && field.startsWith('scores')) {
+              tableRow[`${taskName} - Progress`] = 'not assigned';
+            }
+          });
         }
```

Comment:
For such length `if`/`else` blocks, I would recommend using the early return pattern as it usually makes the code much more readable and maintainable. For example here: 

```js
// If no progressRow is found, mark all scores as "not assigned"
if (!progressRow) {
  scoreReportColumns.value.forEach((column) => {
    const { field, header: taskName } = column; // Use taskName from the column header

    // Ensure field is defined and is a string before calling startsWith
    if (typeof field === 'string' && field.startsWith('scores')) {
      tableRow[`${taskName} - Progress`] = 'not assigned';
    }
  });

  return 
}

scoreReportColumns.value.forEach((column) => {
  const { field, header: taskName } = column; // Use taskName from the column header

  // Ensure field is defined and is a string before calling startsWith
  if (field && typeof field === 'string' && field.startsWith('scores')) {
    // Extract taskId (e.g., "swr", "sre", etc.)
    const scoreKey = field.split('.').slice(-2, -1)[0];

    // Check if taskId exists in progressRow.progress
    tableRow[`${taskName} - Progress`] = progressRow.progress[scoreKey] 
      ? progressRow.progress[scoreKey].value
      : 'not assigned';
  }
});
```